notebook/gateway/managers.py

```diff
@@ -519,6 +519,12 @@ def _get_kernelspecs_endpoint_url(self, kernel_name=None):
 
         return self.base_endpoint
 
+    def _get_endpoint_for_user_filter(self, default_endpoint):
```

Comment:
We should probably make this a static method - particularly since it will be called from the constructor, per previous comment.